crucible-client-types/src/lib.rs

```diff
@@ -71,3 +72,24 @@ impl CrucibleOpts {
         }
     }
 }
+
+/// Display the contents of CrucibleOpts, Only print if keys are populated,
+/// not what the actual contents are.
+impl std::fmt::Display for CrucibleOpts {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Upstairs UUID: {},", self.id)?;
+        write!(f, " Targets: {:?},", self.target)?;
+        write!(f, " lossy: {:?},", self.lossy)?;
+        write!(f, " flush_timeout: {:?},", self.flush_timeout)?;
+        write!(f, " read_only: {:?},", self.read_only)?;
```

Comment:
Is `control` of interest here?

Reply:
It's not.. I think I'm about to make it always on.
But, why not print it till then, so yeah, I'll add it.